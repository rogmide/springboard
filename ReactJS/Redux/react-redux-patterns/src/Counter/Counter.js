import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { INCREMENT, DECREMENT } from "../Actions/actionsType";

function Counter() {
  const count = useSelector((st) => st.count);
  const dispatch = useDispatch();

  function up(evt) {
    dispatch({ type: INCREMENT });
  }
  function down(evt) {
    dispatch({ type: DECREMENT });
  }

  return (
    <div>
      <h2>Count is: {count}</h2>
      <button onClick={up}> + </button>
      <button onClick={down}> - </button>
    </div>
  );
}

export default Counter;
