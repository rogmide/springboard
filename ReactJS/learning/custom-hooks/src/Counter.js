import React from "react";
import useLocalStoreState from "./hooks/useLocalStoreState";

const Counter = () => {
  const [count, setCount] = useLocalStoreState("count", 0);

  const addToCount = () => {
    setCount((count) => count + 1);
  };

  return (
    <>
      <h4>{count}</h4>
      <button onClick={addToCount}>Add</button>
    </>
  );
};

export default Counter;
