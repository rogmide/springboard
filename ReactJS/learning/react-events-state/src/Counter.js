import React, { useState } from "react";

const Counter = () => {
  const [count, setCount] = useState(0);
  return (
    <>
      <h1>Count is: {count}</h1>
      <p>Count</p>
      <button onClick={() => setCount(count + 10)}>Add</button>
      <button onClick={() => setCount(count - 10)}>Subtract</button>
      {/* <input placeholder="username" /> */}
    </>
  );
};

export default Counter;
