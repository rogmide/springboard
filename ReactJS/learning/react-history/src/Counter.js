import React from "react";

class Counter extends React.Component {
  static defaultProps = { color: "green" };
  constructor(props) {
    super(props);
    this.state = { count: 0, isHiding: false };
    // This is ANNOYING
    this.increment = this.increment.bind(this);
    this.decrement = this.decrement.bind(this);
  }

  increment() {
    this.setState({ count: this.state.count + 1 });
  }

  decrement() {
    this.setState({ count: this.state.count - 1 });
  }

  componentDidMount() {
    console.log("Mounted");
  }

  componentDidUpdate() {
    console.log("Updated");
  }

  componentWillUnmount() {
    console.log("Bye Bye");
  }

  render() {
    console.log("Rendering");
    const { color } = this.props;
    return (
      <div>
        <h1 style={{ color }}>I AM COUNTER</h1>
        <h3>Count is: {this.state.count}</h3>
        <button onClick={this.increment}>Add 1</button>
        <button onClick={this.decrement}>Take 1</button>
      </div>
    );
  }
}

export default Counter;
