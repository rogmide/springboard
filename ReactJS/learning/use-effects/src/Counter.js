import React, { useState, useEffect } from "react";

const Counter = () => {
  const [num, setNum] = useState(0);

  const increment = () => {
    setNum((num) => num + 1);
  };

  // This is running after the component is render
  useEffect(() => {
    console.log("RUNNING CALLBACK");
    document.title = `Hi${"!".repeat(num)}`;
  });

  return (
    <div>
      {console.log("RENDERING!!!!")}
      Let's get excited.
      <button onClick={increment}> Add Count </button>
      <p>Counter: {num}</p>
    </div>
  );
};

export default Counter;
