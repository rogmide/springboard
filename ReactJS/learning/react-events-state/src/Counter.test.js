import React from "react";
import { render, screen } from "@testing-library/react";
import Counter from "./Counter";

test("it renders without crashing", () => {
  render(<Counter />);
});

test("it matches snapshot", () => {
  const { asFragment } = render(<Counter />);
  expect(asFragment()).toMatchSnapshot();
});
