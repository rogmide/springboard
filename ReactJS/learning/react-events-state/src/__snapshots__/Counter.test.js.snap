// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Count is: 0
  </h1>
  <button>
    Add
  </button>
  <button>
    Subtract
  </button>
</DocumentFragment>
`;
